app/models/episode_story_handler.rb

```diff
@@ -80,8 +80,12 @@ def update_audio
       []
     end
 
-    episode.media = audio.sort_by(&:position).map do |a|
-      a.links["prx:storage"].href
+    # possible race condition i couldn't quite track down, when replacing files on a
+    # published episode - so let's just not allow deleting it for now
+    unless episode.published? && audio.empty?
+      episode.media = audio.sort_by(&:position).map do |a|
+        a.links["prx:storage"].href
+      end
```

Comment:
Would this allow setting an episode's media via publish UI only one time?  Or another way, the episode should be unpublished first, then update the audio, and then publish the episode again?

Reply:
This only blocks deleting all the files of a published episode.  (I think because of how Publish serializes API calls to CMS, we were briefly announcing an empty `prx:storage` ... causing Feeder to remove the files entirely, instead of replacing them as the user intended).